subprojects/core/src/main/java/org/gradle/api/internal/initialization/DefaultScriptClassPathResolver.java

```diff
@@ -15,37 +15,67 @@
  */
 package org.gradle.api.internal.initialization;
 
+import com.google.common.collect.ImmutableSet;
+import org.gradle.api.Action;
 import org.gradle.api.JavaVersion;
 import org.gradle.api.artifacts.ArtifactView;
 import org.gradle.api.artifacts.Configuration;
-import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.artifacts.ConfigurationContainer;
+import org.gradle.api.artifacts.component.ProjectComponentIdentifier;
 import org.gradle.api.artifacts.dsl.DependencyHandler;
+import org.gradle.api.attributes.Attribute;
 import org.gradle.api.attributes.AttributeContainer;
 import org.gradle.api.attributes.Bundling;
 import org.gradle.api.attributes.Category;
 import org.gradle.api.attributes.LibraryElements;
 import org.gradle.api.attributes.Usage;
 import org.gradle.api.attributes.java.TargetJvmVersion;
 import org.gradle.api.attributes.plugin.GradlePluginApiVersion;
+import org.gradle.api.file.FileCollection;
 import org.gradle.api.internal.artifacts.dsl.dependencies.DependencyFactoryInternal;
+import org.gradle.api.internal.initialization.transform.BaseInstrumentingArtifactTransform;
+import org.gradle.api.internal.initialization.transform.ExternalDependencyInstrumentingArtifactTransform;
+import org.gradle.api.internal.initialization.transform.ProjectDependencyInstrumentingArtifactTransform;
 import org.gradle.api.internal.model.NamedObjectInstantiator;
-import org.gradle.internal.classpath.CachedClasspathTransformer;
+import org.gradle.internal.agents.AgentStatus;
+import org.gradle.internal.classanalysis.AsmConstants;
 import org.gradle.internal.classpath.ClassPath;
 import org.gradle.internal.classpath.DefaultClassPath;
+import org.gradle.internal.classpath.TransformedClassPath;
+import org.gradle.internal.classpath.types.GradleCoreInstrumentingTypeRegistry;
 import org.gradle.internal.component.local.model.OpaqueComponentIdentifier;
 import org.gradle.internal.logging.util.Log4jBannedVersion;
 import org.gradle.util.GradleVersion;
 
+import java.util.EnumSet;
+import java.util.Set;
+
+import static org.gradle.api.artifacts.type.ArtifactTypeDefinition.JAR_TYPE;
+
 public class DefaultScriptClassPathResolver implements ScriptClassPathResolver {
+
+    private static final Set<DependencyFactoryInternal.ClassPathNotation> NO_GRADLE_API = EnumSet.copyOf(ImmutableSet.of(
```

Comment:
:question: why is the intermediate `ImmutableSet` needed? You aren't getting immutability, and there's `EnumSet.of` static factory method to build the set out of values.

Reply:
Good catch, fixed